core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -241,7 +249,53 @@ private Batch filter(final Batch batch) {
         return mergedTags;
     }
 
-    private String selectHost(Metric metric) {
+    /**
+     * Potentially convert some tags to resource identifiers - i.e. tags in tagsToResource conf.
+     *
+     * If there are conflicts, the existing resource identifiers takes precedence over tags.
+     *
+     * @param tags Map of tags that will be used when constructing a metric
+     * @param resource Map of resource identifiers that will be used when constructing a metric
+     */
+    private void processTagsToResource(
+        final Map<String, String> tags, final Map<String, String> resource
+    ) {
+        if (tagsToResource.isEmpty()) {
+            return;
+        }
+
+        tagsToResource.forEach((fromTag, asResource) -> {
```

Comment:
This can be made simpler using the fact that `remove` returns null if the mapping doesn't exist, and we have `putIfAbsent` at our disposal.

```java
final String tag = tags.remove(fromTag);

if (tag != null) {
  resource.putIfAbsent(toResource, tag);
}
```

Reply:
Thanks, fixed now.

Reply:
change not visible in PR